server/images.go

```diff
@@ -48,16 +58,42 @@ type Model struct {
 	ParentModel    string
 	AdapterPaths   []string
 	ProjectorPaths []string
-	Template       string
 	System         string
 	License        []string
 	Digest         string
 	Options        map[string]interface{}
 	Messages       []Message
+
+	Template *template.Template
 }
 
-func (m *Model) IsEmbedding() bool {
-	return slices.Contains(m.Config.ModelFamilies, "bert") || slices.Contains(m.Config.ModelFamilies, "nomic-bert")
+func (m *Model) Can(caps ...Capability) bool {
+	for _, cap := range caps {
+		switch cap {
+		case CapCompletion:
+			f, err := os.Open(m.ModelPath)
+			if err != nil {
+				slog.Error("couldn't open model file", "error", err)
+				continue
+			}
+			defer f.Close()
+
+			ggml, _, err := llm.DecodeGGML(f)
+			if err != nil {
+				slog.Error("couldn't decode ggml", "error", err)
+				continue
+			}
+
+			if _, ok := ggml.KV()[fmt.Sprintf("%s.pooling_type", ggml.KV().Architecture())]; ok {
```

Comment:
Might be a silly question, but are we sure that we won't see this in "hybrid" models at some point?

Reply:
Just want to be careful here since this would turn off text completion

Reply:
pooling is a characteristic of the model that's (currently) only found on embedding models in the gguf context

Reply:
Is there a trait/field we can look for that guarantees text completion capabilities? (vs a looking for the inverse here)

Reply:
not that I know of. gguf kvs aggregate a ton of configurations making it difficult to distinguish one model type from another